debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -270,11 +276,13 @@ private void process(ReplicationMessage message, Long lsn, BlockingConsumer<Chan
             }
         }
 
-        heartbeat.heartbeat(sourceInfo.partition(), sourceInfo.offset(),
-                r -> consumer.accept(new ChangeEvent(r, message.isLastEventForLsn())));
+        if (message.isLastEventForLsn()) {
```

Comment:
Why is the heartbeat emitted only for the last event of an LSN now?

Reply:
LSN flushes for non-complete LSNs are lost so I thought it is better to keep LSN flushes and offset change sin lock-step.

Reply:
Ok, makes sense.